### DISCUSSION TITLE
What does/should credit reservation do?

### DISCUSSION THREAD

**Discussion by DerK96** [alfio-event/alf.io#1104] [Q&A] [answered]
Hi everyone,

when there is a reservation and you go under Pending Payments and pick an ID, there is next to the "Cancel Reservation" button the option to "Credit Reservation".   

What does it do?

I was not able to find it out by myself because in both the demo and a self hosted instance after clicking the button the reservation is kind of stuck. I can provide an issue to that, but i think it would be helpfull for me to know the expected behaviour.

Thank you

**Comment by cbellone** [comment, marked as answer]
Hi @DerK96,

thanks for using alf.io!

The "Credit" button generates a _credit note_ for the reservation. 
Depending on the invoicing regulations of your country of business, you might be required to generate a credit note for each deleted reservation, provided that they have an invoice.

Is that the case? Have you configured alf.io to generate invoices? (see https://alf.io/docs/configuration/invoice/)

If you haven't configured alf.io to generate invoices, you shouldn't see the "credit" button. If you do, that's a bug!

Thanks

**Reply by DerK96** [reply]
Hi @cbellone,

thank you for the fast response.
That helped a lot.

I only configured the setting "_The Country where the organizer runs its Business (can differ from event location)_" under _VAT settings for B2B customers_ and set it to Germany. All other fields in this section were either left blank or set to default.  I guess thats why i was able to see the button. Unfortunately that seems to "stuck" reservations. When trying to modify or delete these i get _Internal Server Error: no items have been updated_ displayed.

After resetting the country, the button disappears as intended.
Regarding the "stuck" Reservations there might be a check implemented if sufficiant information in the invoice section is configured to apply/show the action "Credit Reservation".

Thank you for your help and for providing such a great piece of software.

**Reply by cbellone** [reply]
thank you for the additional investigation

> Regarding the "stuck" Reservations there might be a check implemented if sufficiant information in the invoice section is configured to apply/show the action "Credit Reservation".

yes, indeed. Do you mind opening an issue for this? I believe there is a bug somewhere in that logic

Thank you!
